[Synthetics] Add step detail information to browser monitor alerts

yiannisnikolopoulos opened the pull request:
## Summary

Enhances browser monitor alerts with detailed step information to improve debugging and troubleshooting capabilities. The monitor.step_info field now includes comprehensive context about failed steps, including monitor name, step number, location, and timestamp.

Closes  #156773 

### Before the enhancement

Browser monitor alerts didn't provide information about the steps that failed, making it difficult to understand which specific step failed and when. Users had to manually investigate the Synthetics UI to determine the root cause of failures.

### After the enhancement

Browser monitor alerts now include rich step information in the format:

```

- Step: [Step Number]. [Step Name] - Step script:  [Step Script]

``` 

This provides immediate context about:

- Which step failed (step number and name)
- What was being executed (script source)

### 🔬 How to test

1. Set up a CCS cluster using `edge-lite-oblt` template and start Kibana with `yarn start`
2. Navigate to Synthetics and create a browser monitor with these steps:
```
 step('launch application', async () => {
    await page.goto("https://www.google.com");
  });

  step('assert title', async () => {
    const header = await page.locator('h1');
    expect(await header.textContent()).toBe('todos');
  });
``` 
Use the `North America - US Central` location
4. Set up a `Synthetics monitor status rule` and add a connector (I chose Kibana log) using this message:
```
Monitor "{{context.monitorName}}" is {{{context.status}}} from {{context.locationNames}}.{{{context.pendingLastRunAt}}} - Elastic Synthetics

Details:

- Monitor name: {{context.monitorName}}  
- {{context.monitorUrlLabel}}: {{{context.monitorUrl}}}  
- Monitor type: {{context.monitorType}}  
- Checked at: {{context.checkedAt}}  
- From: {{context.locationNames}}  
- Reason: {{{context.reason}}} 
- Error received: {{{context.lastErrorMessage}}}  
{{{context.failedStepInfo}}}  
{{{context.linkMessage}}}

``` 

5. Wait for the monitor to fail and trigger an alert
6. Check the alert document in Dev Tools:

```
GET .alerts-observability.uptime.alerts*/_search
   {
     "query": {
       "term": {
         "kibana.alert.rule.name": "your-alert-rule-name"
       }
     }
   }
``` 

You can also check this in the alert details `Metadata` tab.

7. Verify that the monitor.step_info field contains the enhanced step information
8. Check that the connector message includes the step information

yiannisnikolopoulos committed:
[Synthetics] Add step detail information to browser monitor alerts

yiannisnikolopoulos committed:
Enhanced step information for browser monitor alerts

- Add synchronous step information fetching in alert creation
- Enhanced step info format with monitor name, step number, location, and timestamp
- Add comprehensive tests for formatStepInformation and getStepInformation functions
- Fix existing tests to handle async scheduleAlert method
- Mock step information fetching in tests to prevent actual ES calls

The step information now displays as:
[Monitor Name] has failed on step [Step Number] [Step Name] in [Location Name] at [HH:MM:SS] on [DD/MM/YYYY].

yiannisnikolopoulos committed:
Merge branch 'main' into add-step-information-to-browser-monitor-alerts

yiannisnikolopoulos committed:
Update translations for Synthetics monitor status messages and fix async tests

yiannisnikolopoulos committed:
fix: update alert actions test to include stepInfo placeholder

Update the expected alert action description in the test to include the {stepInfo} placeholder that was added to support browser monitor step information in alerts.

yiannisnikolopoulos committed:
Add monitor.step_info field to observability_uptime_schema.ts

This field was added to the synthetics rule field map to support step information in browser monitor alerts. The schema needs to be updated to match the field map.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/354591)
* Commit: e6e38ac0e6d21fd003a45ef1389b36ccd910fe20

### Failed CI Steps
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f)
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece)
* [Jest Integration Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8095-414e-84fe-af729b7d0788)
* [FTR Configs #50](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-80c2-44d5-a5e2-90b24cffc741)
* [FTR Configs #101](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-80e3-4542-b3f3-baf2cbd65b28)
* [FTR Configs #101](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16ce-efcd-4237-9325-ca022632db24)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8095-414e-84fe-af729b7d0788) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8095-414e-84fe-af729b7d0788/artifacts/019a16c2-6d11-4bbe-97ed-f2851424a2e3) Jest Integration Tests #<span></span>9 / Alert as data fields checks detect AAD fields changes for: xpack.synthetics.alerts.monitorStatus
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8095-414e-84fe-af729b7d0788) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8095-414e-84fe-af729b7d0788/artifacts/019a16c2-6d11-43f1-bb9b-60f5c5934c76) Jest Integration Tests #<span></span>9 / Alert as data fields checks detect AAD fields changes for: xpack.synthetics.alerts.tls
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8095-414e-84fe-af729b7d0788) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8095-414e-84fe-af729b7d0788/artifacts/019a16c2-6d11-40e1-91a5-2234f0b3051d) Jest Integration Tests #<span></span>9 / Alert as data fields checks detect AAD fields changes for: xpack.uptime.alerts.durationAnomaly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8095-414e-84fe-af729b7d0788) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8095-414e-84fe-af729b7d0788/artifacts/019a16c2-6d10-4c78-ba84-2da82e4f8c43) Jest Integration Tests #<span></span>9 / Alert as data fields checks detect AAD fields changes for: xpack.uptime.alerts.monitorStatus
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8095-414e-84fe-af729b7d0788) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8095-414e-84fe-af729b7d0788/artifacts/019a16c2-6d11-4455-9f58-3d44474ccb97) Jest Integration Tests #<span></span>9 / Alert as data fields checks detect AAD fields changes for: xpack.uptime.alerts.tlsCertificate
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-534b-4e45-accd-03955110209f) Jest Tests #<span></span>11 / message_utils formatStepInformation formats basic step information without context
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a4b-4baa-9d11-be286bd9b46a) Jest Tests #<span></span>11 / message_utils formatStepInformation formats basic step information without context
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51cc-4613-9854-a4bf7c57deb0) Jest Tests #<span></span>11 / message_utils formatStepInformation formats enhanced step information with full context
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a48-4eb3-b54d-e0903186b871) Jest Tests #<span></span>11 / message_utils formatStepInformation formats enhanced step information with full context
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51cb-4760-bd20-0fc18aaaba45) Jest Tests #<span></span>11 / message_utils formatStepInformation handles step action extraction from error message
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a4d-4d0d-ad4e-78e5f6721281) Jest Tests #<span></span>11 / message_utils formatStepInformation handles step action extraction from error message
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-534b-4f41-8f98-ce54a4dd4aaf) Jest Tests #<span></span>11 / message_utils getMonitorSummary handles missing timestamp for pending monitors
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a50-4a20-9bd7-6d97be2cbed8) Jest Tests #<span></span>11 / message_utils getMonitorSummary handles missing timestamp for pending monitors
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-534b-4c27-8c21-a7e6e41493d8) Jest Tests #<span></span>11 / message_utils getMonitorSummary returns correct summary for a ping monitor
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a4d-400e-8600-352e2b532418) Jest Tests #<span></span>11 / message_utils getMonitorSummary returns correct summary for a ping monitor
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-80c2-44d5-a5e2-90b24cffc741) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-80c2-44d5-a5e2-90b24cffc741/artifacts/019a16d3-9202-4259-8c39-5449256d74ac) FTR Configs #<span></span>50 / Serverless Observability feature flag testing - Deployment-agnostic API integration tests SyntheticsCustomStatusRule NumberOfChecks - location threshold = 1 - ungrouped - 1 down location should trigger down for ungrouped
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51cf-4c66-b9af-a22259397fff) Jest Tests #<span></span>11 / setRecoveredAlertsContext handles ungrouped recoveries
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a4c-47b0-bace-0181692cabc5) Jest Tests #<span></span>11 / setRecoveredAlertsContext handles ungrouped recoveries
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-534d-4bc1-b8cf-60301cc13f0f) Jest Tests #<span></span>11 / setRecoveredAlertsContext sets context correctly when location is removed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a50-485b-a14f-16e3be2a87fa) Jest Tests #<span></span>11 / setRecoveredAlertsContext sets context correctly when location is removed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51cd-490d-95c8-09a01c70dcdc) Jest Tests #<span></span>11 / setRecoveredAlertsContext sets context correctly when monitor is deleted
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3c3b-4012-98bb-8493c025cc49) Jest Tests #<span></span>11 / setRecoveredAlertsContext sets context correctly when monitor is deleted
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51d0-47bc-b9bc-151757fc9b85) Jest Tests #<span></span>11 / setRecoveredAlertsContext sets context correctly when monitor is up
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3c3b-4aaf-a383-7ba8ebf5b255) Jest Tests #<span></span>11 / setRecoveredAlertsContext sets context correctly when monitor is up
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51ce-4253-a270-afc5fba01b2b) Jest Tests #<span></span>11 / setRecoveredAlertsContext sets the correct default recovery summary
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a4b-4c75-8e5f-cb64f85e7115) Jest Tests #<span></span>11 / setRecoveredAlertsContext sets the correct default recovery summary
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51cd-437e-b4b7-2f94cec1d72b) Jest Tests #<span></span>11 / setRecoveredAlertsContext sets the recovery summary for recovered custom alerts
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a4c-4d47-8dc8-0c4f8943334e) Jest Tests #<span></span>11 / setRecoveredAlertsContext sets the recovery summary for recovered custom alerts
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-80e3-4542-b3f3-baf2cbd65b28) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-80e3-4542-b3f3-baf2cbd65b28/artifacts/019a16ce-bcb2-4af7-b0f2-d8e02320675d) FTR Configs #<span></span>101 / Stateful Observability feature flag testing  - Deployment-agnostic API integration tests SyntheticsCustomStatusRule NumberOfChecks - location threshold = 1 - ungrouped - 1 down location should trigger down for ungrouped
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16ce-efcd-4237-9325-ca022632db24) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16ce-efcd-4237-9325-ca022632db24/artifacts/019a16d7-e6c8-4387-85a3-b87d7ad3dd95) FTR Configs #<span></span>101 / Stateful Observability feature flag testing  - Deployment-agnostic API integration tests SyntheticsCustomStatusRule NumberOfChecks - location threshold = 1 - ungrouped - 1 down location should trigger down for ungrouped
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51cd-470e-b307-396a1d61081f) Jest Tests #<span></span>11 / StatusRuleExecutor handleDownMonitorThresholdAlert should alert if monitor meet location threshold
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a4d-4ddd-8b28-cba0a5b9274c) Jest Tests #<span></span>11 / StatusRuleExecutor handleDownMonitorThresholdAlert should alert if monitor meet location threshold
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51ce-4ae1-a4fc-0105563ef182) Jest Tests #<span></span>11 / StatusRuleExecutor handleDownMonitorThresholdAlert should send 1 alert for un-grouped
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a4c-417a-b1d5-ceaae7c1e884) Jest Tests #<span></span>11 / StatusRuleExecutor handleDownMonitorThresholdAlert should send 1 alert for un-grouped
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51cc-461f-b62d-a7b1a2033ceb) Jest Tests #<span></span>11 / StatusRuleExecutor handleDownMonitorThresholdAlert should send 2 alerts for un-grouped with 2 different monitors
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a4b-4620-ab59-0e48f9be63e0) Jest Tests #<span></span>11 / StatusRuleExecutor handleDownMonitorThresholdAlert should send 2 alerts for un-grouped with 2 different monitors
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-534b-4e6e-8c36-6c95803655ee) Jest Tests #<span></span>11 / StatusRuleExecutor handlePendingMonitorAlert schedulePendingAlertPerConfigId should group configs by configId and call scheduleAlert with combined location information
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3a4f-49af-9167-9f27d0f1c2b8) Jest Tests #<span></span>11 / StatusRuleExecutor handlePendingMonitorAlert schedulePendingAlertPerConfigId should group configs by configId and call scheduleAlert with combined location information
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16a7-8090-47f9-a15a-c76fad7f304f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16a7-8090-47f9-a15a-c76fad7f304f/artifacts/019a16c0-51cb-42d0-ac6d-9439aaa7fb33) Jest Tests #<span></span>11 / StatusRuleExecutor handlePendingMonitorAlert schedulePendingAlertPerConfigIdPerLocation should call scheduleAlert for each pending config with correct parameters
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/354591#019a16c0-9ed6-48b3-922a-05fff397aece) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/354591/jobs/019a16c0-9ed6-48b3-922a-05fff397aece/artifacts/019a16d9-3c36-48cb-859b-bf31fe524d97) Jest Tests #<span></span>11 / StatusRuleExecutor handlePendingMonitorAlert schedulePendingAlertPerConfigIdPerLocation should call scheduleAlert for each pending config with correct parameters

### History
* :broken_heart: [Build #354016](https://buildkite.com/elastic/kibana-pull-request/builds/354016) failed a99783a40e7b20f3e9ce75c5f28b94c637ad6732
* :broken_heart: [Build #353947](https://buildkite.com/elastic/kibana-pull-request/builds/353947) failed c96bf6ceb7b653ff43223184257800ac34051818

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354591","number":354591,"commit":"e6e38ac0e6d21fd003a45ef1389b36ccd910fe20"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354016","number":354016,"commit":"a99783a40e7b20f3e9ce75c5f28b94c637ad6732"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353947","number":353947,"commit":"c96bf6ceb7b653ff43223184257800ac34051818"}],"number":354591}
buildkite-pr-comment-->

yiannisnikolopoulos committed:
Update failing Jest and Jest Integration tests

yiannisnikolopoulos committed:
fix ungrouped alerts with downConfigs failing FTR test and apply Copilot comments

yiannisnikolopoulos committed:
Merge branch 'main' into add-step-information-to-browser-monitor-alerts

yiannisnikolopoulos committed:
Improve monitor type detection and URL generation for ungrouped alerts

mgiota commented on x-pack/solutions/observability/plugins/synthetics/common/rules/synthetics/translations.ts:
```diff
@@ -12,7 +12,7 @@ export const SyntheticsMonitorStatusTranslations = {
     {
       // the extra spaces before `\n` are needed to properly convert this from markdown to an HTML email
-      defaultMessage: `Monitor "{monitorName}" is {status} from {locationNames}.{pendingLastRunAt} - Elastic Synthetics\n\nDetails:\n\n- Monitor name: {monitorName}  \n- {monitorUrlLabel}: {monitorUrl}  \n- Monitor type: {monitorType}  \n- Checked at: {checkedAt}  \n- From: {locationNames}  \n- Reason: {reason} \n- Error received: {lastErrorMessage}  \n{linkMessage}`,
+      defaultMessage: `Monitor "{monitorName}" is {status} from {locationNames}.{pendingLastRunAt} - Elastic Synthetics\n\nDetails:\n\n- Monitor name: {monitorName}  \n- {monitorUrlLabel}: {monitorUrl}  \n- Monitor type: {monitorType}  \n- Checked at: {checkedAt}  \n- From: {locationNames}  \n- Reason: {reason} \n- Error received: {lastErrorMessage}  \n{stepInfo}{linkMessage}`,
```
@yiannisnikolopoulos I haven't tested it manually yet, but maybe you need to add `\n` after {stepInfo}?

`\n{stepInfo} \n{linkMessage}`

js-jankisalvi reviewed (approved):
response ops changes lgtm 👍

yiannisnikolopoulos committed:
Refactor monitor step information handling in Synthetics alerts

- Renamed 'monitor.step_info' to 'monitor.failed_step_info' across various files to improve clarity and consistency.
- Updated translations in multiple languages to reflect the new field name.
- Adjusted related test cases and utility functions to accommodate the change.
- Enhanced the alert message formatting to include detailed information about failed steps.

This change aims to provide clearer context in alerts regarding failed steps in browser monitors.

yiannisnikolopoulos committed:
Merge branch 'main' into add-step-information-to-browser-monitor-alerts

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/358182)
* Commit: 144b458d636acd29d97e8b2d219d7f2e955b0e42

### Failed CI Steps
* [FTR Configs #84](https://buildkite.com/elastic/kibana-pull-request/builds/358182#019a4f51-10eb-417b-aa65-dcef70ccc4d2)
* [FTR Configs #84](https://buildkite.com/elastic/kibana-pull-request/builds/358182#019a4f7c-174b-4281-90a8-d7e27b837722)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358182#019a4f51-10eb-417b-aa65-dcef70ccc4d2) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358182/jobs/019a4f51-10eb-417b-aa65-dcef70ccc4d2/artifacts/019a4f7b-d5cb-4fcd-9e9c-22ced77d9561) FTR Configs #<span></span>84 / Alerting check alert schemas should not have discrepancies from the alert field map or the field map specific to a rule type
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/358182#019a4f7c-174b-4281-90a8-d7e27b837722) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/358182/jobs/019a4f7c-174b-4281-90a8-d7e27b837722/artifacts/019a4f91-13fd-40d3-83af-6aee72352e5e) FTR Configs #<span></span>84 / Alerting check alert schemas should not have discrepancies from the alert field map or the field map specific to a rule type

### History
* :broken_heart: [Build #356254](https://buildkite.com/elastic/kibana-pull-request/builds/356254) failed 81e6148127429b93426191cfc7ea90ce501f5882
* :broken_heart: [Build #354591](https://buildkite.com/elastic/kibana-pull-request/builds/354591) failed e6e38ac0e6d21fd003a45ef1389b36ccd910fe20
* :broken_heart: [Build #354016](https://buildkite.com/elastic/kibana-pull-request/builds/354016) failed a99783a40e7b20f3e9ce75c5f28b94c637ad6732
* :broken_heart: [Build #353947](https://buildkite.com/elastic/kibana-pull-request/builds/353947) failed c96bf6ceb7b653ff43223184257800ac34051818

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/358182","number":358182,"commit":"144b458d636acd29d97e8b2d219d7f2e955b0e42"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356254","number":356254,"commit":"81e6148127429b93426191cfc7ea90ce501f5882"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354591","number":354591,"commit":"e6e38ac0e6d21fd003a45ef1389b36ccd910fe20"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354016","number":354016,"commit":"a99783a40e7b20f3e9ce75c5f28b94c637ad6732"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353947","number":353947,"commit":"c96bf6ceb7b653ff43223184257800ac34051818"}],"number":358182}
buildkite-pr-comment-->

yiannisnikolopoulos committed:
Regenerate observability_uptime_schema

yiannisnikolopoulos committed:
Merge branch 'main' into add-step-information-to-browser-monitor-alerts

mgiota reviewed (approved):
@yiannisnikolopoulos Did code review only, changes LGTM!

yiannisnikolopoulos committed:
Add step name and number to browser monitor alert subjects and remove step action

yiannisnikolopoulos committed:
Update translations for Synthetics monitor alert subject message

elasticmachine committed:
Merge branch 'main' into add-step-information-to-browser-monitor-alerts

shahzad31 committed:
Merge branch 'main' into add-step-information-to-browser-monitor-alerts

shahzad31 committed:
Merge branch 'main' into add-step-information-to-browser-monitor-alerts

shahzad31 reviewed (approved):
LGTM !!
